tsl/src/continuous_aggs/invalidation_plugin_cache.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#pragma once
+
+#include <postgres.h>
+
+#include <access/htup.h>
+#include <access/tupdesc.h>
+#include <libpq/pqformat.h>
+#include <replication/logical.h>
+#include <utils/hsearch.h>
+
+/*
+ * Invalidation log cache entry.
+ *
+ * This structure is deliberately distinct from InvalidationEntry in
+ * invalidation_record.h, even though they contain similar information.
+ *
+ * The reason is that this structure might add more data that is being tracked
+ * as part of a transaction that should not be present in the record being
+ * sent back from the plugin. If we need to add fields here, it would affect
+ * the format of the records sent back, and we don't what that to happen even
+ * accidentally.
+ */
+typedef struct InvalidationCacheEntry
+{
+	Oid hypertable_relid;
+	int64 lowest_modified;
+	int64 highest_modified;
+} InvalidationCacheEntry;
```

Comment:
Should unify this with struct in `invalidation_record.h` so that we have only one struct with the three fields.

Reply:
I deliberately want to keep these separate since I expect extra entries to be added here for caching and other similar purposes and I don't want these to accidentally "leak" to the record structure.

Reply:
What I suggested is to have a "static" struct for the invalidation records (the triplet). If you want to add extra fields later, either to the cache entry or the msg, you can just embed the static struct in a bigger struct.